HoneycombManager.cs

```diff
@@ -0,0 +1,263 @@
+        nectarM = SMScript.getNectarM();
+        pollenM = SMScript.getPollenM();
+        waxM = SMScript.getWaxM();
+    }
+
+    public float getHCPrice() { return HCPrice; }
+    public int getHCPriceM() { return HCPriceM; }
+
+    void Awake()
+    {
+        HCPriceTxt = HCPriceTxtObj.GetComponent<BuyHoneycombText>();
+        SMScript = StorageManager.GetComponent<StorageManager>();
+        IPScript = InputManager.GetComponent<InputManager>();
+        BGScript = BGManager.GetComponent<HiveBGManager>();
+        CMScript = CameraManager.GetComponent<CameraManager>();
+
+        hcPos = new List<int[]>();
+        honeyStorage = new List<float>();
+        honeyStorageM = new List<int>();
+        nectarStorage = new List<float>();
+        nectarStorageM = new List<int>();
+
+        camera = Camera.main;
+
+        getStorage();
+
+        honeycombNum = 5;
+
+        for(int i = 0; i < honeycombNum; i++)
+        {
+            hcPos.Add(new int[]{i, 0});
+            honeyStorage.Add(0);
+            nectarStorage.Add(0);
+            honeyStorageM.Add(0);
+            nectarStorageM.Add(0);
+        }
+
+        left = 0;
+        right = 6;
+        up = 0;
+        down = 0;
+
+        setupHC();
+    }
+    void Start()
+    {
+        
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        mouseTilePos = IPScript.getMouseTilePos();
+    }
+
+    public void setNectarStorage(int index)
+    {
+        
+    }
+
+    public int getEmptyHCNectar()
+    {
+        for(int i = 0; i < honeycombNum; i++)
+        {
+            if(honeyStorage[i] == 0 && nectarStorage[i] < nectarCapacity)
+            {
+                Debug.Log(i);
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    public Vector3 getHCTilePos(int index)
+    {
+        return BGGridTemp.GetCellCenterWorld(new Vector3Int(hcPos[index][0], hcPos[index][1], 0));
+    }
+
+    public void buyHC()
+    {
+        getStorage();
+
+        if(waxM > HCPriceM || (wax >= HCPrice && waxM == HCPriceM))
+        {
+            Debug.Log("buy");
+
+            hcPos.Add(new int[]{mouseTilePos.x, mouseTilePos.y});
+            honeyStorage.Add(0);
+            nectarStorage.Add(0);
+            honeyStorageM.Add(0);
+            nectarStorageM.Add(0);
+            honeycombNum++;
+
+            wax = wax - HCPrice * (float)Math.Pow(1000, -waxM + HCPriceM);
+            if(wax <= 1)
+            {
+                wax *= 1000;
+                waxM--;
+            }
+
+            SMScript.setWax(wax, waxM);
+            SMScript.changeText();
+
+            BGScript.setupBG();
+
+            if(mouseTilePos.x < left) left = mouseTilePos.x;
+            else if(mouseTilePos.x > right) right = mouseTilePos.x;
+            if(mouseTilePos.y < down) down = mouseTilePos.y;
+            else if(mouseTilePos.y > up) up = mouseTilePos.y;
+
+            Debug.Log(right + " " + left + " " + up + " " + down);
```

Comment:
번거로운 문자열 관리는 문자열 보간 기호 '$'를 이용해 편하게 할 수 있습니다
```suggestion
            Debug.Log($"{right} {left} {up} {down}");
```
[C#의 문자열 보간](https://docs.microsoft.com/ko-kr/dotnet/csharp/tutorials/string-interpolation)